scripts/deploy.md

```diff
@@ -0,0 +1,388 @@
+# Releasing a new version
+
+To release a new version, `deploy.py` is used.
+
+### Prerequisites
+
+`deploy.py` should be run from the repo, directly cloned from
+`LeelaChessZero/lc0` (not fork), and upstreams from `master` and `release`
+branches should be configured to point to that repo.
+
+It's recommended to have a separate repository cloned which you will use
+exclusively for releases, and won't do other development on that repo.
+
+To push version changes, scripts needs write access to this repo.
+
+
+### General script information
+
+There are three pipelines implemented in this script:
+- `new_minor_release`
+- `cherrypick`
+- `new_patch_release`
+
+Every pipeline consists of series of steps.
+
+To start pipeline, run it like this:
+```bash
+$ ./deploy.py new_patch_release
+```
+
+To just get a list of steps in a pipeline, do that:
+```bash
+$ ./deploy.py --list new_patch_release
+```
+
+To start pipeline from particular step (5 in this example), do that:
+```bash
+$ ./deploy.py --start 5 new_patch_release
+```
+
+To make pipeline ask confirmation before executing every step, run like this:
+```bash
+$ ./deploy.py --steps new_patch_release
+```
+
+The state of the pipeline is stored in file `/tmp/<pipeline-name>_state.json`, 
+so if pipeline crashes (or you Ctrl+C it), when you restart pipeline with
+the same parameters, it will offer you to continue from where you stopped.
+
+That's the recommended way of handling stuff when something is wrong:
+Ctrl+C, then fix stuff manually, then restart and continue.
+
+
+## Pipelines
+
+### New minor/major release
+
+When issuing new minor/major release, entire contents of `master` branch is
+merged into `release` branch, new minor version change happens, and `-rc1`
+suffix is added.
+
+The version in `master` branch is after that updated to a next minor version,
+with `dev` suffix added.
+
+Examples:
+
+- `v0.13.1 -> v0.14-rc1`, master branch becomes `v0.15-dev`
```

Comment:
Without having yet reviewed the python, postfixes must come after the patch version. `0.14-rc1` isn't a valid version, but `0.14.0-rc1` is. Ditto `0.15-dev` no bueno, needs must be `0.15.0-dev`